Gitweb:     http://git.kernel.org/linus/7d1849aff6687a135a8da3a75e32a00e3137a5e2
Commit:     7d1849aff6687a135a8da3a75e32a00e3137a5e2
Parent:     6d20792e85187b27ae3d1b76678a2dd7025e8bc2
Author:     Mikael Pettersson <mikpe@it.uu.se>
AuthorDate: Thu Dec 3 15:52:44 2009 +0100
Committer:  Ingo Molnar <mingo@elte.hu>
CommitDate: Thu Dec 3 16:25:15 2009 +0100

    x86, apic: Enable lapic nmi watchdog on AMD Family 11h
    
    The x86 lapic nmi watchdog does not recognize AMD Family 11h,
    resulting in:
    
      NMI watchdog: CPU not supported
    
    As far as I can see from available documentation (the BKDM),
    family 11h looks identical to family 10h as far as the PMU
    is concerned.
    
    Extending the check to accept family 11h results in:
    
      Testing NMI watchdog ... OK.
    
    I've been running with this change on a Turion X2 Ultra ZM-82
    laptop for a couple of weeks now without problems.
    
    Signed-off-by: Mikael Pettersson <mikpe@it.uu.se>
    Cc: Andreas Herrmann <andreas.herrmann3@amd.com>
    Cc: Joerg Roedel <joerg.roedel@amd.com>
    Cc: <stable@kernel.org>
    LKML-Reference: <19223.53436.931768.278021@pilspetsen.it.uu.se>
    Signed-off-by: Ingo Molnar <mingo@elte.hu>
---
 arch/x86/kernel/cpu/perfctr-watchdog.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/x86/kernel/cpu/perfctr-watchdog.c b/arch/x86/kernel/cpu/perfctr-watchdog.c
index fab786f..898df97 100644
--- a/arch/x86/kernel/cpu/perfctr-watchdog.c
+++ b/arch/x86/kernel/cpu/perfctr-watchdog.c
@@ -712,7 +712,7 @@ static void probe_nmi_watchdog(void)
 	switch (boot_cpu_data.x86_vendor) {
 	case X86_VENDOR_AMD:
 		if (boot_cpu_data.x86 != 6 && boot_cpu_data.x86 != 15 &&
-		    boot_cpu_data.x86 != 16)
+		    boot_cpu_data.x86 != 16 && boot_cpu_data.x86 != 17)
 			return;
 		wd_ops = &k7_wd_ops;
 		break;
